Examples/test-suite/preproc_defined.i

```diff
@@ -123,3 +123,30 @@ void another_macro_checking(void) {
 #if 0
 # wobble wobble
 #endif
+
+#define __attribute__(x)
+%inline %{
+/*
+ * This should not compile in when this bug is present
+ * https://github.com/swig/swig/issues/2525
+ */
+#define my_bad_attr __attribute__
+#define my_good_attr(x) __attribute__(x)
+
+int bad my_bad_attr((used));
+int good my_good_attr((used));
+
+/*
+ * The behaviour of Self-Referential Macros is defined
+ * https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Self-Referential-Macros.html
+ */
+int y = 0;
+
+#define x (4 + y)
+#define y (2 * x)
+
+int z = y;
```

Comment:
This testcase fails because this isn't valid C code:

```
#define x (4 + y)
#define y (2 * x)

int z = y;
$ gcc -Wall -W -c t.c
t.c:4:11: error: initializer element is not constant
    4 | #define y (2 * x)
      |           ^
t.c:6:9: note: in expansion of macro ‘y’
    6 | int z = y;
      |         ^
```

The line expands to `int z = (2 * (4 + y));` and `y` isn't constant.

I'll see if I can come up with something that works without having to switch this to be a C++ testcase, but seriously you really need to be making sure the tests pass **locally** before pushing changes or else you're wasting everybody's time (including your own).